### PR TITLE
Zijie lint fix for reports component Phase 8

### PR DESCRIPTION
**Pull request by iTvX**
# Description
Lint fix for Reports component, this is phase 8.

## Files changed
All js/jsx file in /src/components/Reports/TasksDetail

eslint test:
CleanShot 2023-09-30 at 18 50 33@2x

# How to Test

To test this pull request:

- Check out the current branch
- Run `npm install` and `npm run start:local` to run this PR locally  
- Log in as an admin user
- Check all the functionality of the report is the same as before, and see if any bugs have appeared. (Please record a video)

(Optional)
- Set up ESLint in VSCode/ IntelliJ IDEA, I personally use IntelliJ IDEA
- Delete .eslintignore file in the root: `rm .eslintignore`
- `npx eslint --fix src/components/Reports/TableFilter/index.js`
- `npx eslint --fix src/components/Reports/TableFilter/TableFilter.jsx`
- `git restore .`

I have complete confidence that **the code will pass the ESLint tests.** Therefore, I kindly request that you **focus on testing the frontend**. Specifically, please ensure that the **report functionality remains consistent** with how it operated previously. Thank you for your time and effort!

### CODE REVIEW

**Comment by beblicarl** [comment]
The Report component works like it used to. Great work !!

[screencast-localhost_3000-2023.10.01-14_26_03.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/25bbc511-989e-44ef-8e32-732a11fcaeb0)

**Review by beblicarl — APPROVED** [review]

**Comment by RheaWu1212** [comment]
Works really great for this PR!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/109626741/a62df3dc-b8c7-4ffc-a4ed-369a3a206305

**Comment by Vishala09** [comment]
The eslint command shows no error!
image

Tested the reports component. 
All of the below works:
- Selecting category displays the corresponding list on right
- Selecting a Filter works properly
- Search with a keyword works properly
- Selecting the corresponding project/people/team from the list opens their details fine
- The date filter works fine
- Show details after the date filter displays the details well
[Untitled_ Oct 02,2023 4_39 PM.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/65274029/22279876-beda-48fd-a9ee-606a591b9711)

**Comment by iTvX** [comment]
> The eslint command shows no error! image
> 
> Tested the reports component. All of the below works:
> 
> * Selecting category displays the corresponding list on right
> * Selecting a Filter works properly
> * Search with a keyword works properly
> * Selecting the corresponding project/people/team from the list opens their details fine
> * The date filter works fine
> * Show details after the date filter displays the details well
> 
> Please find my video testing here :
> 
>  Untitled_.Oct.02.2023.4_39.PM.webm

The reason showing that is because you didn't delete eslint ignore file.
Delete .eslintignore file in the root: `rm .eslintignore`

**Review by Vishala09 — DISMISSED** [review]

**Comment by malikjahanzaib** [comment]
Running ESLint shows no errors on either of the files: "src/components/Reports/TableFilter/index.js" and "src/components/Reports/TableFilter/TableFilter.jsx".

Screenshot:
image

Test Recording:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46221334/289e4034-00a6-4be0-be97-9dbaabe1037a

Tests for the Report Component performed:

- Category Filter: Displays contents correctly.
- Active/Inactive Filter: Works Correctly.
- Date Filter: Works Correctly.
- Reports: Works Correctly.
- Weekly Summaries: Works Correctly.

**Review by malikjahanzaib — APPROVED** [review]

**Comment by olenadanykh** [comment]
hi @iTvX. 
No ESLint errors for me. The page works as expected, but when I inspect it, I notice an error in the console, same one as in Phase 9 PR. I have recorded a video for more details. Overall, great work.
PR1363

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49422649/4b684899-b933-4a97-8f1c-689856e7a1a1

**Review by DavidC0126 — DISMISSED** [review]
Hi @iTvX I tested this PR and it works as intended! Good job.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/08829e85-f85c-4dce-baae-8aeb97c60342

**Review by Sssky-18 — APPROVED** [review]
Hi @iTvX, I tested the reports page using your branch in this PR, and it worked normally with some minor warnings, which I think should be irrelevant to this PR. The buttons of projects, people, and teams worked properly and all the information displayed as usual with correct UI. The search button also worked as intended, which was able to filter everything based on providing text or date. Therefore, I just approved it and thank you for your contribution!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73833864/74e7f923-c36f-4193-a221-ba340895e89c

**Review by Jiyuanxia — DISMISSED** [review]
I have tested your pull request and it works as same as dev branch. Good job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92963742/9c894815-6da3-4138-a87c-fe2f7396c352

**Review by Shubhu0 — APPROVED** [review]
Hi @iTvX Tested the PR and it is working as intended though I did notice that there are some areas where it is slow! Noticeably during the total team graphs and specific team numbers! Please refer to video for the noticed issue


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56580102/215a7cc2-af1b-40ec-ad79-08f02b558348

**Review by savio-henrique — APPROVED** [review]
Hi @iTvX , I've tested the code as you've asked, no errors in the ESLint and working as intended on the frontend!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83360232/d3bc843a-d13b-459a-a531-9b5f72a08187

terminal

**Review by yi-lin-1234 — APPROVED** [review]
Hello, I tested your PR and works as intended. Great work!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92185695/0607e28b-ab9a-4471-b18b-1542fce27236



Screenshot 2023-10-25 at 10 27 06 AM

**Review by Chuehleo — APPROVED** [review]
Everything works as expected, great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/122568562/d79aea6f-7649-42ec-bb1d-c54b365d15a0

File: src/components/Reports/TableFilter/TableFilter.jsx [hunk 1]
```diff
@@ -1,13 +1,14 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (EvianTan) [inline src/components/Reports/TableFilter/TableFilter.jsx:1]: Same comment as [here](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1356/files#r1384348686)

**Comment by KurtisIvey** [comment]
+1 to actually fixing the linting issue rather than disabling it. 100% in agreement with EvianTan. 

Once the issue EvianTan brought up is addressed, this can move to final review

File: src/components/Reports/TableFilter/TableFilter.jsx (hunk 1 above)
Comment (iTvX) [reply src/components/Reports/TableFilter/TableFilter.jsx]: fixed

**Review by EvianTan — APPROVED** [review]

**Pull request merged**